sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -637,6 +637,33 @@ class QueryExecutionErrorsSuite
       sqlState = "0A000")
   }
 
+  test("FAILED_RENAME_PATH: rename when destination path already exists") {
+    var srcPath: Path = null
+    var dstPath: Path = null
+
+    val e = intercept[SparkFileAlreadyExistsException](
+      withTempPath { p =>
+        val basePath = new Path(p.getAbsolutePath)
+        val fm = new FileSystemBasedCheckpointFileManager(basePath, new Configuration())
+        srcPath = new Path(s"$basePath/src")
+        fm.createAtomic(srcPath, overwriteIfPossible = true).close()
+        dstPath = new Path(s"$basePath/dst")
+        fm.createAtomic(dstPath, overwriteIfPossible = true).close()
+
+        fm.renameTempFile(srcPath, dstPath, false)
```

Comment:
Could we add this case through the user perspective API

Reply:
@LuciferYang I couldn't think of a simple way of triggering this error through the user-facing API. I think the only code that raises it is in [CheckpointFileManager](https://github.com/apache/spark/blob/b84ddd5e71acc9ae3facdf47148becef3861d11d/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala#L268). The destination path is [generated randomly](https://github.com/apache/spark/blob/b84ddd5e71acc9ae3facdf47148becef3861d11d/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala#L224) and not accessible by any user-facing API I could find.

Do you have a suggestion for how to trigger this error through the user-facing API?

Reply:
You could use a fake file system to say that a file exists already, see
https://github.com/apache/spark/blob/91af7e8f562461bfdb686f383a0811a28772acf7/sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala#L426

Reply:
Thanks for those suggestions. I changed the test to use the user API. Please review again.